gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/IncrementalConfigurationCachePlugin.java

```diff
@@ -146,6 +151,16 @@ private static void createDirectories(Path directory) {
     }
 
     private void configureTaskCompatibility(Project project, Set<String> lockListTasks) {
+        // We need a way to always run with configuration cache for every task. This is needed as in some cases the
+        // allow list and its lock file will be out of sync. In these cases we still want the
+        // dryRunConfigurationCacheEnabledTasks task to all its dry-run tasks with configuration cache, so that all
+        // issues (lock file out of date and tasks that are not CC friendly) are surfaced at once.
+        if (project.getProviders()
+                .gradleProperty("configuration-cache-compatible-for-all-tasks")
+                .isPresent()) {
```

Comment:
think it might just be easier to do `project.hasProperty("configuration-cache-compatible-for-all-tasks")`